circuits/cpp/scripts/run_coverage

```diff
@@ -0,0 +1,78 @@
+#!bin/bash
+set -e
+
+# To be called LOCALLY for testing WITHOUT docker.
+#
+# Contains core logic for collecting test coverage in the current environment.
+#
+# Run from circuits/cpp/
+# Example:
+# ./scripts/run_coverage glob
+#
+# Running with the CLEAN environment variable set will delete the build-coverage folder and start again.
+# Running with the CLEAR_COV environment variable set will clear existing vscode coverage data (lcov.info).
+#
+# This will create two outputs: lcov.info and an html coverage report in build-coverage/coverage_report
+#
+# Note: Assumes ignition SRS transcripts have already been downloaded
+
+# TESTS:
+# 1. names of test executables to run
+#   - (like "aztec3_circuits_kernel_tests aztec3_circuits_abis_tests")
+# 2. or "glob" (LOWERCASE) to run all built tests
+# 3. or a file containing a list of test
+TESTS=$1
+
+# Set coverage build directory
+BUILD_DIR="build-coverage"
+
+# If the clean file is set, wipe all existing coverage data
+if [[ -n "$CLEAN" ]]; then
+  echo "The CLEAN environment variable is set. Clearing existing coverage data."
+  rm -rf ./$BUILD_DIR || true
+fi
+
+# Perform coverage build
+echo "Setting cmake to run with preset coverage"
+cmake -Bbuild-coverage -DCMAKE_BUILD_TYPE=Debug -DUSE_TURBO=true --preset=coverage .
```

Comment:
Oh we are.... We shouldn't need `-Bbuild-coverage -DCMAKE_BUILD_TYPE=Debug` since those should be in the `coverage` preset like bberg. If they aren't then that should be fixed.